scripts/pdf/render.py

```diff
@@ -15,106 +15,66 @@
-    title_content = "<h1 class=title-main>tldr pages</h1>" \
+    # A string that stores all pages in HTML format
+    html = '<!doctype html><html><head><meta charset="utf-8">' \
+        +"</head><body><h1 class=title-main>tldr pages</h1>" \
         + "<h4 class=title-sub>Simplified and community-driven man pages</h4>" \
         + "<h6 class=title-sub><em><small>Generated on " + datetime.now().strftime("%c") + "</small></em></h6>" \
-        + "</body></html>"
-
-    # Creating title page
-    with open("title.html", "w") as f:
-        f.write(header + title_content)
+        + '<p style="page-break-before: always" ></p>'
 
-    group.append(HTML("title.html").render())
+    # Writing names of all directories inside 'pages' to a list
+    for os_dir in os.listdir(loc):
+        oslist.append(os_dir)
+    oslist.sort()
 
     for operating_sys in oslist:
-
         # Required string to create directory title pages
-        dir_title = "<h2 class=title-dir>" + \
-            operating_sys.capitalize() + "</h2></body></html>"
-
-        # Creating directory title page for current directory
-        with open("dir_title.html", "w") as os_html:
-            os_html.write(header + dir_title)
-
-        group.append(HTML("dir_title.html").render())
+        html += "<h2 class=title-dir>" + operating_sys.capitalize() + "</h2>" \
+            + '<p style="page-break-before: always" ></p>'
 
         # Creating a list of all md files in the current directory
         for temp in glob.glob(os.path.join(loc, operating_sys, "*.md")):
             allmd.append(temp)
-
-        # Sorting all filenames in the directory, to maintain the order of the PDF
         allmd.sort()
 
-        # Conversion of Markdown to HTML
+        # Conversion of Markdown to HTML string
         for page_number, md in enumerate(allmd, start=1):
-
-                with open(md, "r") as inp:
-                    text = inp.readlines()
-
-                with open("htmlout.html", "w") as out:
-                    out.write(header)
-
-                    for line in text:
-                        if re.match(r'^>', line):
-                            line = line[:0] + '####' + line[1:]
-                        html = markdown.markdown(line)
-                        out.write(html)
-                    out.write(footer)
-
-                group.append(HTML("htmlout.html").render())
-                print("Rendered page {} of the directory {}".format(
-                    str(page_number), operating_sys))
-
+            with open(md, "r") as inp:
+                text = inp.readlines()
+                for line in text:
+                    if re.match(r'^>', line):
+                        line = line[:0] + '####' + line[1:]
+                    html += markdown.markdown(line)
+            html += '<p style="page-break-before: always" ></p>'
+            print("Rendered page {} of the directory {}".format(
+                str(page_number), operating_sys))
```

Comment:
Given we're using 3.9 to run this script, might as well make this a f-string while we're making changes here:

```suggestion
            print(f"Rendered page {page_number} of the directory {operating_sys}")
```